rdr_service/tools/tool_libs/consent_validation_report.py

```diff
@@ -99,63 +107,51 @@
                    (cf.is_signature_valid and NOT cf.is_signing_date_valid) AS invalid_signing_date,
                    -- Invalid DOB conditions: DOB missing, DOB before defined cutoff, DOB in the future, or
                    -- DOB later than the consent authored date
-                   (p.date_of_birth is null or p.date_of_birth < "{dob_cutoff}"
-                    or p.date_of_birth > "{report_date}")
-                    or (p.consent_for_study_enrollment_authored is not null
-                        and p.date_of_birth > p.consent_for_study_enrollment_authored )
-                       AS invalid_dob,
-                    -- Invalid age at consent error only applies to PRIMARY consent type
-                   (cf.type = 1 and TIMESTAMPDIFF(YEAR, p.date_of_birth, p.consent_for_study_enrollment_authored) < 18)
-                      AS invalid_age_at_consent,
+                   (ps.date_of_birth is null or ps.date_of_birth < "{dob_cutoff}" or ps.date_of_birth > "{report_date}")
+                    or (ps.consent_for_study_enrollment_authored is not null
+                        and ps.date_of_birth > ps.consent_for_study_enrollment_first_yes_authored )
+                    AS invalid_dob,
+                   TIMESTAMPDIFF(YEAR, COALESCE(ps.date_of_birth, CURRENT_DATE),
+                                 ps.consent_for_study_enrollment_first_yes_authored) < 18
+                    AS invalid_age_at_consent,
                    -- Map the text for other errors we know about to its TRACKED_CONSENT_ERRORS name
                    (cf.file_exists AND cf.other_errors LIKE '%missing consent check mark%') AS checkbox_unchecked,
                    (cf.file_exists AND cf.other_errors LIKE '%non-veteran consent for veteran participant%')
                       AS non_va_consent_for_va,
                    (cf.file_exists AND cf.other_errors LIKE '%veteran consent for non-veteran participant%')
                       AS va_consent_for_non_va
-            FROM consent_file cf
-            JOIN participant_summary p on p.participant_id = cf.participant_id
-            LEFT OUTER JOIN hpo h on p.hpo_id = h.hpo_id
-            LEFT OUTER JOIN organization o on p.organization_id = o.organization_id
+            FROM participant_summary ps
+            -- Eliminate test/ghost pids from the result
+            JOIN participant p on p.participant_id = ps.participant_id
+                 AND p.is_test_participant = 0 and (p.is_ghost_id is null or not p.is_ghost_id) and p.hpo_id != 21
```

Comment:
Should the email and phone number also be looked at?

Reply:
Personally, I'd like to get away from that because I think it masks cases where PTSC should have flagged test participants for us and didn't.  Would like for them to be forced to look at them, where the resolution would be for them to properly flag them via the process they agreed to follow?

Reply:
Sounds good to me.